packages/client/src/forms/configuration/configFields/utils.ts

```diff
@@ -10,70 +10,229 @@
  * graphic logo are (registered/a) trademark(s) of Plan International.
  */
 
-import { Event, IFormField, IFormSection, IForm } from '@client/forms'
-
-export type IConfigFormField = {
-  fieldId: string
-  precedingFieldId: string | null
-  foregoingFieldId: string | null
-  required: boolean
-  enabled: string
-  custom: boolean
-  definition: IFormField
+import {
+  Event,
+  IFormField,
+  IForm,
+  IQuestionConfig,
+  IFormSection,
+  ISerializedFormSection,
+  ISerializedFormSectionGroup,
+  IFormSectionGroup,
+  SerializedFormField
+} from '@client/forms'
+import { deserializeFormField } from '@client/forms/mappings/deserializer'
+import { createCustomField } from '@client/forms/configuration/customUtils'
+import { FieldPosition } from '@client/forms/configuration'
+import { FieldEnabled } from '@client/forms/configuration/defaultUtils'
+import { registerForms } from '@client/forms/configuration/default'
+
+export type IDefaultConfigField = Pick<
+  IQuestionConfig,
+  'fieldId' | 'enabled' | 'required' | 'preceedingFieldId' | 'custom'
+> & {
+  foregoingFieldId: string
+  identifiers: {
+    sectionIndex: number
+    groupIndex: number
+    fieldIndex: number
+  }
+}
+
+type ICustomConfigField = IQuestionConfig & {
+  foregoingFieldId: string
 }
 
-type IFormFieldMap = Record<string, IConfigFormField>
+export type IConfigField = IDefaultConfigField | ICustomConfigField
 
-export type ISectionFieldMap = Record<string, IFormFieldMap>
+export type IConfigFieldMap = Record<string, IConfigField>
 
-export function getContentKey(field: IConfigFormField) {
-  return field.definition.label.id
+export type ISectionFieldMap = Record<string, IConfigFieldMap>
+
+function defaultFieldToQuestionConfig(
+  fieldId: string,
+  preceedingFieldId: string,
+  sectionIndex: number,
+  groupIndex: number,
+  fieldIndex: number,
+  field: IFormField
+): IDefaultConfigField {
+  return {
+    fieldId,
+    enabled: field.enabled ?? '',
+    required: field.required,
+    preceedingFieldId,
+    foregoingFieldId: FieldPosition.BOTTOM,
+    identifiers: {
+      sectionIndex,
+      groupIndex,
+      fieldIndex
+    }
+  }
 }
 
-export function getCertificateHandlebar(field: IConfigFormField) {
-  return field.definition.mapping?.template?.[0]
+function customFieldToQuestionConfig(
+  fieldId: string,
+  preceedingFieldId: string,
+  field: IFormField
+): ICustomConfigField {
+  const messageProperties = [
+    'label',
+    'placeholder',
+    'tooltip',
+    'description'
+  ] as const
+  return {
+    fieldId,
+    preceedingFieldId,
+    enabled: field.enabled ?? '',
+    foregoingFieldId: FieldPosition.BOTTOM,
+    ...messageProperties.reduce(
+      (accum, prop) => ({
+        ...accum,
+        [prop]: [{ lang: 'en', descriptor: field[prop] }]
+      }),
+      {}
+    )
+  }
```

Comment:
@yeasinhossain-dsi Should I perhaps use the default language for `lang`?

Reply:
i would say yes